Featurestats

PJBrs opened the pull request:
This PR aims to clean up some code in feature application, adds more racial spells automatically, adds class-based darkvision, and adds some feat bonuses to initiative.

For review, I'd like to point out two things:
1. I removed all spell stuff from race.py
2. Some races include first level spells at second level, which I implemented as follows:

```python
+    class HellishRebukeAtLvl2(spells.HellishRebuke):
+        level = 2
+        __doc__ = spells.HellishRebuke.__doc__.replace('2nd', '3rd').replace('1st', '2nd')
+        name = "Hellish Rebuke (Racial Feature)"
+
+    _spells = {1: [spells.Thaumaturgy,], 3: [HellishRebukeAtLvl2,]}
+    spells_known = []
+    spells_prepared = []

```

PJBrs commented:
I noticed that this PR was sitting here for a while already, which made me have another look at these patches. On second thought, I think I was wrong in the way I tried to implement, e.g., Hellish Rebuke at level 2, in the sense that that wasn't according to the rules. And on closer inspection so many of these racial feature spells follow specific rules, that perhaps it is easier to just add them and let the player closely read their feats. If you want, I can squash the first and fourth patches.

canismarko commented:
Yeah sorry, a lot of these PRs accomplish more than one thing which makes it time-consuming to review and I've been very busy lately for reasons that aren't worth getting into here. I really do appreciate you putting in the considerable time to contribute these improvements.

1e7d69d19b8557ccc6dc2d3c68c96ea441046ded and e01e9e9269491dfc056e64524d70d7e57d8c07fc are good.

You are proposing undoing cc7be9d139fce8bb9d51e68b5e44e1fb4602bf2e and 440236b29dbaa1d1b649643c7e709ad0180a3bad, right?

I'm fine with that. If you want to push commits that do that, I will merge this PR as is.

PJBrs commented:
> You are proposing undoing [cc7be9d](https://github.com/canismarko/dungeon-sheets/commit/cc7be9d139fce8bb9d51e68b5e44e1fb4602bf2e) and [440236b](https://github.com/canismarko/dungeon-sheets/commit/440236b29dbaa1d1b649643c7e709ad0180a3bad), right?

No, not quite,  [440236b](https://github.com/canismarko/dungeon-sheets/commit/440236b29dbaa1d1b649643c7e709ad0180a3bad) is a partial undo of [cc7be9d](https://github.com/canismarko/dungeon-sheets/commit/cc7be9d139fce8bb9d51e68b5e44e1fb4602bf2e). But I can see that perhaps in these two patches I wanted to accomplish a little bit to much. I'll change it a bit and explain better....

PJBrs commented:
So, I split the most complex patch in this series. The two patches that you wanted to merge in any case are:
https://github.com/canismarko/dungeon-sheets/pull/159/commits/7727c1385c995e464ad07a142bd0bf0a65ffedf1
and
https://github.com/canismarko/dungeon-sheets/pull/159/commits/e005acdf9da635969b0ec2b4bd38ec572499faf3

The rest of this PR I now split in three different patches. The first one (https://github.com/canismarko/dungeon-sheets/pull/159/commits/443a5afa359985e3e4d0a705bf97c77d805594be) adds more racial spells.

The final two patches together first move any remaining spells from race.py to features/race,py, and then remove all spell-related code from race.py.

Let me know if this works better for you. If not, just tell me which patches to remove!

Also, thanks for taking the time to review this, and the best of luck dealing with those busy times!

PJBrs commented:
@canismarko  Anything else I can do for this PR?
